src/coreclr/vm/gcinfodecoder.cpp

```diff
@@ -2183,6 +2183,35 @@ template <typename GcInfoEncoding> void TGcInfoDecoder<GcInfoEncoding>::ReportRe
 
 #endif // Unknown platform
 
+#ifdef FEATURE_INTERPRETER
+template <> OBJECTREF* TGcInfoDecoder<InterpreterGcInfoEncoding>::GetStackSlot(
+                        INT32           spOffset,
+                        GcStackSlotBase spBase,
+                        PREGDISPLAY     pRD
+                        )
+{
+    OBJECTREF* pObjRef = NULL;
+
+    if( GC_SP_REL == spBase )
+    {
+        _ASSERTE(!"GC_SP_REL is invalid for interpreter frames");
+    }
+    else if( GC_CALLER_SP_REL == spBase )
+    {
+        _ASSERTE(!"GC_CALLER_SP_REL is invalid for interpreter frames");
+    }
+    else
+    {
+        _ASSERTE( GC_FRAMEREG_REL == spBase );
```

Comment:
If we only ever need a single stack base in the interpreter, we can skip encoding it in the GCInfo and just use `NO_STACK_BASE_REGISTER`.

Reply:
Do you know how I would do that? SetStackBaseRegister doesn't accept NO_STACK_BASE_REGISTER as an argument. I'm not explicitly setting a stack base register anywhere, so shouldn't I get the right behavior as a default?

Reply:
There is stack base register for the method, and stack base register for each slot. This is about the stack base register for each slot.

The GCInfo encoder/decoder is hardcode to use two bits for the per slot base register: https://github.com/dotnet/runtime/blob/dd75c45c123055baacd7aa4418f425f412797a29/src/coreclr/vm/gcinfodecoder.cpp#L1267 . Skipping this for interpreter would require a bunch of plumbing that's probably not worth it.

Reply:
(I thought that it is abstracted via gcinfotypes.h when I wrote this suggestion, but it is apparently not the case.)

Reply:
I'll leave a TODO comment for the sake of future improvement, thanks!